### PR TITLE
Implement early exit threshold config entry

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes https://github.com/contour-terminal/contour/issues/1460

### CODE REVIEW

File: src/contour/ContourGuiApp.cpp [hunk 1]
```diff
@@ -149,7 +149,17 @@ crispy::cli::command ContourGuiApp::parameterDefinition() const
 
 std::chrono::seconds ContourGuiApp::earlyExitThreshold() const
 {
-    return std::chrono::seconds(parameters().get<unsigned>("contour.terminal.early-exit-threshold"));
+    auto const configThreshold = config().earlyExitThreshold.value();
+    auto const parameterThreshold = parameters().get<unsigned>("contour.terminal.early-exit-threshold");
+
+    // default threshold is config::documentation::DefaultEarlyExitThreshold seconds
+    if (parameterThreshold != config::documentation::DefaultEarlyExitThreshold)
+        return std::chrono::seconds(parameterThreshold);
```
Comment (christianparpart) [inline src/contour/ContourGuiApp.cpp:157]: what if the user actually explicitly passed an early-exit-threshold that matches the default? Then this branch would not be taken.

File: src/contour/contour.yml [hunk 2]
```diff
@@ -64,6 +64,9 @@ pty_buffer_size: 1048576
 
 default_profile: main
 
+# Time in seconds to check for early threshold
+early_exit_threshold: 5
+
```
Comment (christianparpart) [inline src/contour/contour.yml:69]: Didn't we make this contour.yml file obsolete?

Comment (Yaraslaut) [reply src/contour/contour.yml:69]: We do not use it to generate config file, this is true, i just placed it there to keep it updated until we have a better way to see full default config file

File: src/contour/ContourGuiApp.cpp (hunk 1 above)
Comment (Yaraslaut) [reply src/contour/ContourGuiApp.cpp]: You are right, updated implementation

**Review by christianparpart — APPROVED** [review]

**Pull request merged**